service/ExchangeService.js

```diff
@@ -10,19 +10,22 @@ var log = new Logger(AppConfig.logLevel);
 
 var fetchExchangeData = function(exchange, url, callback){
 	request.get(url, function (err, response, body) {
-		if ( !err && response.statusCode == 200 ){
+				if ( !err && response.statusCode == 200 ){
 			try {
 				callback(null, JSON.parse(body));
 			}catch (e) {
-				log.debug('ERROR parsing response from ' + exchange + '. Details: ' + e + '\nResponse::' + body);
-				return callback(null, null);
+				var parsingError = 'ERROR parsing response from ' + exchange + '. Details: ' + e + '\nResponse::' + body;
+				log.debug(parsingError);
+				return callback(new Error(parsingError), null);
```

Comment:
If you are worried about input sanitization, one thing you could do is just `new Error("Error parsing JSON")`, but I will let one of the JS experts comment on this.